File: solns/clojure/22.clj
(require '[clojure.string :as s])

(def data
  (s/split (s/replace (slurp "../data/22.txt") #"\"" "") #","))

(def scores
  (zipmap (map char (range (int \A) (inc (int \Z))))
          (range 1 (inc 26))))

(defn score [x]
  (reduce + (map scores x)))

(def m
  (zipmap (range 1 (inc (count data)))
          (map score (sort data))))

(def result
  (reduce +
    (for [k (keys m)]
      (* k (m k)))))

(println result)
(assert (= result 871198282))

File: solns/ruby/22.rb
def score(name)
  scores = ('A'..'Z').zip(1..26).to_h
  name.chars.map { |c| scores[c] }.inject(&:+)
end

file = File.new('/data/22.txt')
data = file.gets.gsub('"', '').split(',') # All one line in file
file.close

result = data.each_index
             .zip(data.sort)
             .map { |t| (t.first + 1) * score(t.last) }
             .inject(&:+)

puts result
raise Error unless result == 871_198_282
